frontend/document/koptinterface.lua

```diff
@@ -1239,6 +1238,26 @@ function KoptInterface:getPageBoxesFromPositions(doc, pageno, ppos0, ppos1)
     end
 end
 
+--[[--
+Compare positions within one page.
+Returns 1 if positions are ordered (if ppos2 is after ppos1), -1 if not, 0 if same.
+Actually positions of the word boxes containing ppos1 and ppos2 are compared.
+--]]
+function KoptInterface:comparePositions(doc, ppos1, ppos2)
+    local box1 = self:getWordFromPosition(doc, ppos1).pbox
+    local box2 = self:getWordFromPosition(doc, ppos2).pbox
+    if box1.y == box2.y then
```

Comment:
Dunno what ppos1 looks like, it doesn't have any `y` ? Or does it have a `y` but it may not be "current" ?

Reply:
`comparePositions` function is used in ReaderBookmark and in ReaderHighlight (extended selection).
In ReaderHighlight one of the positions is the `hold_pos`, so at first we need to get a word under the long-press and then compare.

Reply:
OK, so you would need this only when you invoke comparePositions() from extendSelection() ?
When you invoke comparePositions() from isBookmarkInPositionOrder(), you know you have bookmarks that are not random screen coordinates, but are proper word boxes ?
So, may be it could have another parameter to say we need to go thru this or not?

Reply:
Sorry, missed your comment.
pos0 and pos1 stored in pdf bookmarks are not box positions, but touch positions.
We always need to get a word and its box's x and y.

Reply:
That's a good idea, pboxes are ordered and it is easy to get coordinates from them and compare.
We already have pboxes in some bookmarks (when adding a note, to save it in the pdf file).

In the meantime, currently pboxes are stored in highlights, but used only when saving highlight to pdf. If this option is disabled, pboxes are not needed.
Readerview does not use them to draw the highlights, just positions
https://github.com/koreader/koreader/blob/7163a392a8e6bf6ed580eae5a8b746025d1baf45/frontend/apps/reader/modules/readerview.lua#L484-L485